core/Nonce.php

```diff
@@ -99,6 +99,22 @@ public static function verifyNonce($id, $cnonce, $expectedReferrerHost = null)
         return true;
     }
 
+
+    /**
+     * Returns if a nonce is valid md5.
+     *
+     * @param string $md5
+     * @return bool
+     */
+    public static function isValidMd5($md5)
+    {
+        if (preg_match('/^[a-f0-9]{32}$/', $md5)) {
```

Comment:
Should we also check for capital `A-F`?

Reply:
You are right, capital letters are valid hex codes too. I can make the string lowercase before the `preg_match` check.

Reply:
probably ctype_xdigit would work too